core/src/main/java/tech/tablesaw/analytic/AnalyticQuery.java

```diff
@@ -129,44 +171,74 @@ public String toSqlString() {
 
   @Override
   public String toString() {
-    return toSqlString();
+    return toSqlLikeString();
   }
 
+  /**
+   * Executes the query adding all the calculated columns to a new table. The result columns will
+   * have the same order as the from table.
+   *
+   * @return a new table containing only the result columns.
+   */
   public Table execute() {
     return AnalyticQueryEngine.create(this).execute();
   }
 
+  /**
+   * Executes the query and adds all the calculated columns directly to the source table.
+   *
+   * @throws IllegalArgumentException if any of the calculated columns have the same name as one of
+   *     the columns in the FROM table.
+   */
+  public void executeInPlace() {
+    // Throw an exception if there will be duplicate columns before executing the query.
+    Set<String> existingColumnNames =
+        this.table.columnNames().stream().map(String::toLowerCase).collect(Collectors.toSet());
+    Set<String> newColumnNames =
```

Comment:
nit, i would leave off the _this._ where it isn't needed.

Reply:
done